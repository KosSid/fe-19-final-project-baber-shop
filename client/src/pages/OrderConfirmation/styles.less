@import '~antd/lib/style/themes/default.less';

.order-confirmation {
  .ant-btn-primary {
    color: @white;
    background: @black;
    border: 1px solid @black;
  }
}