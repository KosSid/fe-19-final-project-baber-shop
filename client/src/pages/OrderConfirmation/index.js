import React from 'react';
import {Button, Result} from "antd";
import {Link} from "react-router-dom";
import './styles.less';

const OrderConfirmation = () => {

    return (
        <div className="order-confirmation">
            <Result
                status="success"
                title="Order has been placed!"
                extra={
                    <Link to="/">
                        <Button className='to-basket-button' type={'primary'}>Go to Homepage</Button>
                    </Link>
                }
            />
        </div>
    )
}

export default OrderConfirmation;