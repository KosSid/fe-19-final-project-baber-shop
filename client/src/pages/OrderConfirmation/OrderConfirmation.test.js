import React from 'react'
import {render, unmountComponentAtNode} from 'react-dom'
import {act} from '@testing-library/react'
import OrderConfirmation from "./index";
import {MemoryRouter} from "react-router";

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
})

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
})

test('OrderConfirmation is rendered true', () => {
  act(() => {
    render(<MemoryRouter>
      <OrderConfirmation/>
    </MemoryRouter>, container)
  });

  expect(container.innerHTML).toMatchSnapshot();
})