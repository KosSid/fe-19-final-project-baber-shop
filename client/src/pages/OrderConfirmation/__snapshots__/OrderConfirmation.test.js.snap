// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderConfirmation is rendered true 1`] = `"<div class=\\"order-confirmation\\"><div class=\\"ant-result ant-result-success\\"><div class=\\"ant-result-icon\\"><span role=\\"img\\" aria-label=\\"check-circle\\" class=\\"anticon anticon-check-circle\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"check-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z\\"></path></svg></span></div><div class=\\"ant-result-title\\">Order has been placed!</div><div class=\\"ant-result-extra\\"><a href=\\"/\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-primary to-basket-button\\"><span>Go to Homepage</span></button></a></div></div></div>"`;
